### PR TITLE
Snowplow Snowflake Loader 0.5.0 blog post

### PR DESCRIPTION
**Pull request by aldemirenes**

### CODE REVIEW

File: _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+<!--more-->
+
+<h2 id="bad-row">1. Bad Row Support</h2>
+
+Data quality has always been a huge focus for Snowplow. Being a non-lossy data pipeline is one of the central piece of this data quality puzzle. Indeed, other pieces such as data validity (through our schema validation technology) or data richness (through our enrichments) connect directly with the non-lossiness piece.
+
+In practice, for a Snowplow pipeline, non-lossiness meant that when something went wrong anywhere in the pipeline, instead of discarding it, the data impacted is parked as “bad” for later inspection, fixing and reprocessing. In Snowplow jargon, this bad data is called “bad rows”.
+
+With this release, we are adding bad row support to Snowflake Transformer with [its new format][new-bad-row-format-rfc]. From now on, Snowflake Transformer will continue to run when it encounters with unexpected data and it will write the bad row to separate place for further inspection instead of halting with exception.
```
Comment (chuwy) [inline _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md]: Can we also mention that it is the first component of the pipeline introducing bad rows and that enrichment is about to be next one very soon?

**Comment by chuwy** [comment]
Hey @LyubaGolovina! Could you review it please for us?

File: _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+You have to specify where you want to store the bad rows in the config file. You can find detailed information about it in the [upgrading part](#upgrading).
+
+<h2 id="kryo-serialization">2. Kryo serialization</h2>
+
+With this release, we are starting to use [Kryo][kryo] library instead of Java serialization to serialize objects more quickly. This will bring performance improvement to Snowflake Transformer job.
```
Comment (chuwy) [inline _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md]: > instead of Java serialization

Technically, this is not entirely accurate. 

I'd rephrase this whole paragraph to be more like "Other changes" and mention that we've made improvements in serializiation which gave us X% performance boost.

File: _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md (hunk 1 above)
Comment (LyubaGolovina) [inline _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md]: Edit to: From now on, Snowflake Transformer will continue to run when it encounters unexpected data and it will write the bad rows to a separate place for further inspection instead of halting with exception.

**Comment by chuwy** [comment]
Hey @aldemirenes, sorry for a short notice, but could you please also mention iglu URIs for a bad rows schemas? Users would like to check how bad rows look like.

**Comment by aldemirenes** [comment]
@LyubaGolovina @chuwy I've made some changes after your comments. Could you go over it one more time ?

File: _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+<!--more-->
+
+<h2 id="bad-row">1. Bad Row Support</h2>
+
+Data quality has always been a huge focus for Snowplow. Being a non-lossy data pipeline is one of the central pieces of this data quality puzzle. Indeed, other pieces such as data validity (through our schema validation technology) or data richness (through our enrichments) connect directly with the non-lossiness piece.
+
+In practice, for a Snowplow pipeline, non-lossiness meant that when something went wrong anywhere in the pipeline, instead of discarding it, the data impacted was parked as “bad” for later inspection, fixing and reprocessing. In Snowplow jargon, this bad data is called “bad rows”.
+
+With this release, we are adding bad row support to Snowflake Transformer with [its new format][new-bad-row-format-rfc]. From now on, Snowflake Transformer will continue to run when it encounters unexpected data and it will write the bad rows to a separate place for further inspection instead of halting with exception. Snowflake Loader is the first component of the pipeline introducing bad rows and enrichment is about to be next one very soon.
```
Comment (LyubaGolovina) [inline _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md]: The last sentence in this section is a bit unclear, does it mean bad rows will soon be introduced to enrichments as well? In this case, it would be better to say "Snowflake Loader is the first component of the pipeline introducing bad rows, and soon enrichment will also include this feature." or something along these lines.

File: _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+In order to make bad rows easy to understand, we separated them to two types as loader parsing error and Snowflake error. As it can be understood from names, loader parsing error represensts cases where enriched event can not be parsed to [Snowplow Event][event-model] successfully and Snowflake error represents cases where enriched event parsed without problem however something went wrong internally while trying to transform the event to suitable format for Snowflake. You can look at [schema of loader parsing error][loader-parsing-error-schema] and [schema of Snowflake error][snowflake-error-schema] if you want to learn more about their format.
```
Comment (LyubaGolovina) [inline _posts/2019-06-18-snowplow-snowflake-loader-0.5.0-released/index.md]: I'd recommend editing this sentence to read: "In order to make bad rows easy to understand, we separated them into two types, loader parsing error and Snowflake error. As the names imply, loader parsing error represents cases where enriched events can not be parsed to [Snowplow Event][event-model] successfully and Snowflake error represents cases where enriched events are parsed without problem, however something went wrong internally while trying to transform the event into a suitable format for Snowflake.

**Comment by aldemirenes** [comment]
Hey @LyubaGolovina! Could you take a look at it one last time before release ?

**Comment by chuwy** [comment]
Apparently Lyuba is off until next week. Having that new changes are super-small ([1](https://github.com/snowplow/snowplowanalytics.com/compare/ac5285ef722701f1c6d5fd6d42e4d23ae709dde3..76347919ca1c19605e5d037723b56ea44e72553e), [2](https://github.com/snowplow/snowplowanalytics.com/compare/76347919ca1c19605e5d037723b56ea44e72553e..3ca6b3aba7bca2db7c0b9f3974148453cda31f01)) I'm wondering if @franciskadfyi or @nepsil0n can review/approve them?

File: _posts/2019-08-28-snowplow-snowflake-loader-0.5.0-released/index.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+<!--more-->
+
+<h2 id="bad-row">1. Bad Row Support</h2>
+
+Data quality has always been a huge focus for Snowplow. Being a non-lossy data pipeline is one of the central pieces of this data quality puzzle. Indeed, other pieces such as data validity (through our schema validation technology) or data richness (through our enrichments) connect directly with the non-lossiness piece.
+
+In practice, for a Snowplow pipeline, non-lossiness meant that when something went wrong anywhere in the pipeline, instead of discarding it, the data impacted was parked as “bad” for later inspection, fixing and reprocessing. In Snowplow jargon, this bad data is called “bad rows”.
+
+With this release, we are adding bad row support to Snowflake Transformer with [its new format][new-bad-row-format-rfc]. From now on, Snowflake Transformer will continue to run when it encounters unexpected data and it will write the bad rows to a separate place for further inspection instead of halting with exception. Snowflake Loader is the second component of the pipeline introducing new bad row format after [RDB Loader][rdb-loader-31].
```
Comment (franciskadfyi) [inline _posts/2019-08-28-snowplow-snowflake-loader-0.5.0-released/index.md]: Maybe rephrase: "Snowflake Loader is the second component of the pipeline introducing new bad row format after" to 

"Snowflake Loader is the second component of the pipeline to introduce the new bad row format after"

**Comment by franciskadfyi** [comment]
Hi @aldemirenes and @chuwy - I just suggested a minor change (take it or leave). Otherwise I think this is ready to go. Let us know and we'll get this live!

**Comment by nepsil0n** [comment]
Released this from different branch along with another post that does not contain the current conflicts. We can review and publish to prod.
